src/interactable_019.c

```diff
@@ -0,0 +1,332 @@
+// anything beyond Leaf Forest Act 2 will use a wrong AnimID.
+static const u16 sInt019_AnimationIds[]
+    = { [LEVEL_INDEX(ZONE_1, ACT_1)] = SA2_ANIM_PLATFORM_LF_WIDE,
+        [LEVEL_INDEX(ZONE_1, ACT_2)] = SA2_ANIM_PLATFORM_LF_WIDE };
+
+extern const struct SpriteTables *gUnknown_03002794;
+
+extern u32 sub_800C060(struct UNK_0808B3FC_UNK240 *, s32, s32, struct SomeStruct_59E0 *);
+
+static void Task_Interactable_019(void);
+
+static void Task_805E35C(void);
+static void Task_805E480(void);
+static void Task_805E6A4(void);
+static void TaskDestructor_Interactable019(struct Task *);
+
+// @TODO: Replace with tile-count from the graphics data itself
+#define IA_019_NUM_TILES 32
+
+void initSprite_Interactable_019(Interactable *ia, u16 spriteRegionX, u16 spriteRegionY,
+                                 u8 spriteY)
+{
+    struct Task *t = TaskCreate(Task_Interactable_019, sizeof(Sprite_019), 0x2000, 0,
+                                TaskDestructor_Interactable019);
+    Sprite_019 *platform = TaskGetStructPtr(t);
+    SpriteBase *base = &platform->base;
+    struct UNK_0808B3FC_UNK240 *displayed = &platform->displayed;
+    // s16 screenX, screenY;
+
+    base->regionX = spriteRegionX;
+    base->regionY = spriteRegionY;
+    base->ia = ia;
+    base->spriteX = ia->x;
+    base->spriteY = spriteY;
+    SET_SPRITE_INITIALIZED(ia);
+
+    platform->unk3C = 0;
+
+    // @BUG Loads the -2 set through SET_SPRITE_INITIALIZED
+    displayed->unk16 = SpriteGetScreenPos(ia->x, spriteRegionX);
+    displayed->unk18 = SpriteGetScreenPos(ia->y, spriteRegionY);
+    displayed->unk4 = VramMalloc(IA_019_NUM_TILES);
+
+#ifdef UBFIX
+    // Prevent overflow
+    displayed->unkA
+        = sInt019_AnimationIds[gCurrentLevel % ARRAY_COUNT(sInt019_AnimationIds)];
+#else
+    displayed->unkA = sInt019_AnimationIds[gCurrentLevel];
+#endif
+    displayed->unk20 = 0;
+    displayed->unk1A = 1152;
+    displayed->unk8 = 0;
+    displayed->unk14 = 0;
+    displayed->unk1C = 0;
+
+    displayed->unk21 = 0xFF;
+    displayed->unk22 = 0x10;
+    displayed->unk25 = 0;
+    displayed->unk10 = 0x2000;
+
+    if (ia->d.sData[0] != 0) {
+        displayed->unk10 |= 0x400;
+    }
+
+    sub_8004558(displayed);
+}
+
+void Task_Interactable_019(void)
+{
+    Sprite_019 *platform = TaskGetStructPtr(gCurTask);
+    SpriteBase *base = &platform->base;
+    struct UNK_0808B3FC_UNK240 *displayed = &platform->displayed;
+    Interactable *ia = base->ia;
+    s16 screenX, screenY;
+
+    screenX = SpriteGetScreenPos(base->spriteX, base->regionX);
+    screenY = SpriteGetScreenPos(ia->y, base->regionY);
+
+    displayed->unk16 = screenX - gCamera.unk0;
+    displayed->unk18 = screenY - gCamera.unk4;
+
+    if (sub_800C060(displayed, screenX, screenY, &gPlayer) & 0x8) {
+        gCurTask->main = Task_805E35C;
+        platform->unk3C = 0;
+    }
+
+    if ((gGameMode >= GAME_MODE_MULTI_PLAYER)
+        && ((s8)ia->x == SPRITE_STATE_UNK_MINUS_THREE)) {
```

Comment:
x = state?

Reply:
Yep. It's a mechanism they use. That's why there's stuff like this all over all interactables:
https://github.com/freshollie/sa2/blob/interactables/src/interactable_019.c#L121

Reply:
But why is the x value considered state at all? It's this just a position value not state? Or do I misunderstand?

Reply:
Cos if it comes from `spriteX` then is that not a position?

Reply:
The x position is received when spawning, so the game knows where inside a "sprite region" it should spawn at.
While it's active, the x's value gets used as a state of some kind. I only know that (-2) stands for "initialized", but no idea what (-3) is.
But when it's supposed to be destroyed, the state gets reset to its x-value, so that the next time it spawns, it does so at the expected position.

Reply:
It probably shouldn't be named `->x` then?